src/dawn/CodeGen/GridTools/GTCodeGen.cpp

```diff
@@ -793,32 +796,47 @@ void GTCodeGen::generateStencilClasses(
         // is resolved
         // https://github.com/MeteoSwiss-APN/dawn/issues/110
         // ===-----------------------------------------------------------------------------------===
-        for(int dim = 0; dim < ext.getSize() - 1; ++dim) {
-          std::string at_call = "template at<" + std::to_string(dim) + ">()";
-          std::string storage = codeGenProperties.getParamType(stencilInstantiation, fieldInfo);
-          // assert for + accesses
-          // ===---------------------------------------------------------------------------------===
-          // PRODUCTIONTODO: [STAGGERING]
-          // we need the staggering offset in K in order to have valid production code
-          // https://github.com/MeteoSwiss-APN/dawn/issues/108
-          // ===---------------------------------------------------------------------------------===
-          std::string staggeringoffset = (dim == 2) ? " - 1" : "";
-          StencilConstructor.addStatement(
-              "static_assert((static_cast<int>(" + storage + "::storage_info_t::halo_t::" +
-              at_call + ") >= " + std::to_string(ext[dim].Plus) + staggeringoffset + ") || " + "(" +
-              storage + "::storage_info_t::layout_t::" + at_call + " == -1)," +
-              "\"Used extents exceed halo limits.\")");
-          // assert for - accesses
-          StencilConstructor.addStatement("static_assert(((-1)*static_cast<int>(" + storage +
-                                          "::storage_info_t::halo_t::" + at_call + ") <= " +
-                                          std::to_string(ext[dim].Minus) + ") || " + "(" + storage +
-                                          "::storage_info_t::layout_t::" + at_call + " == -1)," +
-                                          "\"Used extents exceed halo limits.\")");
+        std::string storage = codeGenProperties.getParamType(stencilInstantiation, fieldInfo);
+        if(fieldsExtsMap.find(storage) == fieldsExtsMap.end()) {
```

Comment:
usually in maps you use find (with it's return value) if you want to use the iterator and count otherwise. so for what you're doing here there are two ways - either works fine:
```
if(!fieldsExtsMap.count(storage)) {
  fieldsExtsMap.insert(std::make_pair(storage, ext));
} else {
  fieldsExtsMap[storage].merge(ext);
}
```
or
```
auto element = fieldsExtsMap.find(storage);
if(element == == fieldsExtsMap.end()) {
    fieldsExtsMap.emplace(storage, ext); 
} else {
    (*element).second.merge(ext);
}
```

Reply:
now using second alternative